kernel/src/main/java/org/cristalise/kernel/events/History.java

```diff
@@ -76,6 +76,19 @@ public Event addEvent(AgentPath agentPath, AgentPath delegatePath, String agentR
         return storeNewEvent(newEvent);
     }
 
+    public Event addEvent(AgentPath agentPath, AgentPath delegatePath, String agentRole,
+                          String stepName, String stepPath, String stepType, Schema schema, 
+                          StateMachine stateMachine, int transitionId, String viewName, byte[] attachment)
```

Comment:
please use the boolean here. this makes the method signature far more consistent.